Framework/Algorithms/src/XDataConverter.cpp

```diff
@@ -103,7 +103,7 @@ void XDataConverter::setXData(API::MatrixWorkspace_sptr outputWS,
         }
       }
     }
-    outputWS->setX(index, m_cachedX);
+    outputWS->setSharedX(index, m_cachedX);
   } else {
     const MantidVec &xBoundaries = inputWS->readX(index);
     MantidVec &xPoints = outputWS->dataX(index);
```

Comment:
You can change `ConvertToPointData::calculateXPoints` to take `HistogramX` as arguments. Then either use the `rawData()` accessor to get a vector, or change `convertToBinCentre` (there are just a few algorithms that use it).